src/main/java/com/minecraftabnormals/abnormals_core/common/advancement/modification/AdvancementModificationManager.java

```diff
@@ -101,7 +102,14 @@ private static TargetedAdvancementModifier deserializeModifiers(JsonElement json
 			if (config == null) {
 				throw new JsonParseException("Missing 'config' element!");
 			}
-			advancementModifiers.add(modifier.deserialize(config, conditionArrayParser));
+			boolean conditionsMet = true;
+			if (JSONUtils.hasField(entry, "conditions")) {
+				if (!CraftingHelper.processConditions(JSONUtils.getJsonArray(entry, "conditions"))) {
+					AbnormalsCore.LOGGER.info("Skipped advancement modifier \"" + type + "\" for advancement \"" + advancement + "\" as its conditions were not met");
+					conditionsMet = false;
+				}
+			}
+			if (conditionsMet) advancementModifiers.add(modifier.deserialize(config, conditionArrayParser));
```

Comment:
```suggestion
			if (!JSONUtils.hasField(entry, "conditions") || CraftingHelper.processConditions(JSONUtils.getJsonArray(entry, "conditions"))) {
			                advancementModifiers.add(modifier.deserialize(config, conditionArrayParser));
			} else {
			      AbnormalsCore.LOGGER.info("Skipped advancement modifier \"" + type + "\" for advancement \"" + advancement + "\" as its conditions were not met");
			}
```
This would be cleaner. The formatting might be a bit messed up.